chainer/initializers/__init__.py

```diff
@@ -20,6 +20,30 @@
 HeUniform = uniform.HeUniform
 
 
+def generate_array(initializer, shape, xp):
+    """Return initialized array.
+
+    The algorithms used to make the new values depend on the
+    concrete derived classes.
```

Comment:
Please note that the `dtype` of a generated array depends on `initializer.dtype`.